### PR TITLE
moving Mapstyle selector to Option menu with select2 and rebased from #2040

### PR DESCRIPTION
**Pull request by callmebackdraft**
*added dropdown menu to Option menu with map styles in select2 style
*made the dropdown read localstorage so it reads the last used style
*made it compatible with local with .json files
*removed mapstyle selector from map itself

More screen real estate especially on mobile
 also responds to #1674

## Motivation and Context
real estate upgrade especially on small screens (see issue: #1674)

## How Has This Been Tested?
after editing loaded page on:
 Microsoft Edge 
 Chrome ( Win10)
 IPhone Safari
 IPhone Chrome

## Screenshots (if appropriate):
image

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by andrewda** [comment]
I don't know, I kinda like the native Google Maps design.

File: static/js/map.js [hunk 1]
```diff
@@ -1151,6 +1152,40 @@ $(function() {
 });
 
 $(function() {
+    // populate Navbar Style menu
+    $selectStyle = $("#map-style")
+	
+	// Load Stylenames from locale and populate lists
+	$.getJSON("static/locales/mapstyle." + language + ".json").done(function(data){
+        var styleList = []
```
Comment (chuyskywalker) [inline static/js/map.js]: nit pick: the indentation in this update is a bit off.

File: static/js/map.js [hunk 2]
```diff
@@ -1151,6 +1152,40 @@ $(function() {
...
+        $selectStyle.select2({
+            placeholder: "Select Style",
+            data: styleList
+        });
+
+        // setup the list change behavior
+        $selectStyle.on("change", function (e) {
+            selectedStyle = $selectStyle.val();
+            map.setMapTypeId(selectedStyle)
+        Store.set('map_style', selectedStyle)
```
Comment (chuyskywalker) [inline static/js/map.js]: Indentation and `;`'s please.

**Comment by chuyskywalker** [comment]
Functionally working. I'm on board with this selector getting pushed out of the way -- it's rarely used and doesn't need to take up UI space with such prominence. 

There are some spacing, code quality items to take care of, but functionally and for it's purpose I'm a :+1:

File: templates/map.html [hunk 3]
```diff
@@ -147,6 +147,13 @@
             </label>
           </div>
         </div>
+        <div class="form-control">
+            <hr width="100%">
+            <label for="map-style">
+                <h3>Style</h3>
```
Comment (chuyskywalker) [inline templates/map.html]: Personal preference would be "Map Style" instead of just "Style"

**Comment by callmebackdraft** [comment]
@chuyskywalker thank you for your feedback, i fixed the indentation and agreed with you on the naming convention.

**Comment by chuyskywalker** [comment]
Wait for one more collab to give this a go and sign off, thanks!

File: templates/map.html [hunk 4]
```diff
@@ -147,6 +147,13 @@
             </label>
           </div>
         </div>
+        <div class="form-control">
+            <hr width="100%">
+            <label for="map-style">
+                <h3>Map Style</h3>
+                <select id="map-style" style="min-width:100%;"></select>
+            </label>
+        </div>
```
Comment (andrewda) [inline templates/map.html]: still need to fix spacing here (2 spaces per indentation level)

**Comment by andrewda** [comment]
Alright, fine, I see the appeal with this. After the spacing issue above is fixed, :+1: and merge.

**Comment by callmebackdraft** [comment]
@andrewda done

**Comment by willgrnr** [comment]
Do we really need a text box for that drop down? It's annoying for mobile users as it brings up the keyboard (which is a pain in the ass with the sidebar as it is) -  there's not enough options to warrant typing imo.

**Pull request merged**